Source/RESTyard.ContractFirst/RESTyard.Generator/Templates/client/typescript.sbn

```diff
@@ -28,38 +28,51 @@ export function match<TIn, TOut>(nullable: Nullable<TIn>, onValue: (v: TIn) => T
     }
 }
 {{
-    for parametersType in TransferParameters
-        parameters = []
-        for property in parametersType.Property
+    func gatherParameterArguments
+        parametersRetVal = []
+        checkProperties $0.typeName $0.Property
+        for property in $0.Property | array.filter @isVisibleProperty
             param = property.name + ": " + (mapNullableType (!property.mandatory) property.type)
-            parameters = array.add parameters param
+            parametersRetVal = array.add parametersRetVal param
+        end
+        ret parametersRetVal
+    end
+~}}
+{{
+    for parametersType in TransferParameters
+        parameters = gatherParameterArguments parametersType
+        parentParameters = []
+        currentType = parametersType
+        while isNotEmpty currentType.parentType
+            currentType = findParentParameters TransferParameters currentType.parentType
+            parentParameters = array.concat (gatherParameterArguments currentType) parentParameters
         end
+        allParameters = array.concat parentParameters parameters
 }}
-export class {{ parametersType.typeName }} {
+export class {{ parametersType.typeName }}{{ if isNotEmpty parametersType.parentType }} extends {{ parametersType.parentType }} {{ end }}{
     constructor(
-{{- for param in parameters }}
+{{- for param in allParameters }}
         public readonly {{ param }}{{ if !for.last }},{{ end }}
 {{- end }}
-    ) { }
+    ) { {{ if (array.size parentParameters) != 0 }}super({{- for param in parentParameters}}{{ param | string.split ':' | array.first }}{{ if !for.last }}, {{ end }}{{- end}});{{ end }} }
```

Comment:
Apparently in typescript `super()` needs to be called when extending types so this is wrong if the parent has no parameters.